packages/server/src/app.module.ts

```diff
@@ -1,9 +1,34 @@
+import * as path from 'path';
 import { Module } from '@nestjs/common';
+import { ConfigModule, ConfigService } from '@nestjs/config';
+import { MongooseModule } from '@nestjs/mongoose';
 import { AppController } from './app.controller';
 import { AppService } from './app.service';
 
 @Module({
-  imports: [],
+  imports: [
+    ConfigModule.forRoot({
+      envFilePath:
+        process.env.ENV_FILE || path.join(__dirname, '../../../.env'),
+    }),
+    MongooseModule.forRootAsync({
+      imports: [ConfigModule],
+      useFactory: async (configService: ConfigService) => {
+        let uri = configService.get<string>('DB_URL');
+        if (!uri) {
+          uri = `mongodb://${configService.get<string>(
+            'MONGO_ADMIN_USR',
+          )}:${configService.get<string>(
+            'MONGO_ADMIN_PWD',
+          )}@localhost:${configService.get<string>('MONGO_PORT')}`;
+        }
```

Comment:
I think this if is not needed. In local in your .env file you could have

```
DB_URL="mongodb://${MONGO_ADMIN_USR}:${MONGO_ADMIN_PWD}@localhost:${MONGO_PORT}"
```
so I don't think this logic is needed

Reply:
Before doing it, Ive tried on several ways of reusing the env variables in it, but always got it the literal strings

![image](https://user-images.githubusercontent.com/1570567/100387215-d3c9af00-3027-11eb-97d6-9d3297f194e9.png)

Reply:
It's apparently a limitation from dotenv package (used behind the scenes).
We could, though, use https://github.com/motdotla/dotenv-expand to expand those.

Let's go with this approach?
( I am not a fan of rebuilding the string on the server either)

Reply:
AH!
Nevermind, there is an option for it on the ConfigModule from nestjs.

I am updating it right away